src/main/ruby/truffleruby/core/truffle/io_operations.rb

```diff
@@ -543,10 +543,14 @@ def self.normalize_options(mode, perm, options, default_mode = nil)
             external, internal = encoding.split(':', 2)
           end
         end
+
+        unless Primitive.nil? options[:path]
+          path = StringValue(options[:path])
```

Comment:
`path = options[:path]` above this `unless` would avoid two hash lookups in case it is set